Enable public MEC on CAPZ (#2836)
kubernetes-sigs/cluster-api-provider-azure: docs/book/src/topics/publicmec-clusters.md

# Deploy cluster on Public MEC

- **Feature status:** Experimental
- **Feature gate:** EdgeZone=true

## Overview

> Cluster API Provider Azure (CAPZ) experimentally supports deploying clusters on [Azure Public MEC](https://azure.microsoft.com/en-us/solutions/public-multi-access-edge-compute-mec). Before you begin, you may need to prepare an azure subscription which has access to Public MEC.

Reviewer: ```suggestion
Cluster API Provider Azure (CAPZ) has experimental support for deploying clusters on [Azure Public MEC](https://azure.microsoft.com/en-us/solutions/public-multi-access-edge-compute-mec). Before you begin, you need an Azure subscription which has access to Public MEC.
```

> To deploy a cluster on Public MEC, provide extendedlocation info through environment variable, and use the Edgezone flavor.

Reviewer: ```suggestion
To deploy a cluster on Public MEC, provide extended location info through environment variables and use the "edgezone" flavor.
```

## Example: Deploy cluster on Public MEC by `clusterctl`

> A clusterctl flavor exists to deploy cluster on Public MEC. This flavor requires the following environment variables to be set before executing clusterctl.

Reviewer: ```suggestion
The clusterctl "edgezone" flavor exists to deploy clusters on Public MEC. This flavor requires the following environment variables to be set before executing `clusterctl`.
```

```bash
# Kubernetes values
export CLUSTER_NAME="my-cluster"
export WORKER_MACHINE_COUNT=2
export CONTROL_PLANE_MACHINE_COUNT=1
export KUBERNETES_VERSION="v1.25.0"

# Azure values
export AZURE_LOCATION="eastus2euap"
export AZURE_EXTENDEDLOCATION_TYPE="EdgeZone"
export AZURE_EXTENDEDLOCATION_NAME="microsoftrrdclab3"
export AZURE_RESOURCE_GROUP="${CLUSTER_NAME}"
```

Create a new service principal and save to local file:
```bash
az ad sp create-for-rbac --role Contributor --scopes="/subscriptions/${AZURE_SUBSCRIPTION_ID}" --sdk-auth > sp.json
```

> export the following variables in you current shell.

Reviewer: ```suggestion
Export the following variables to your current shell.
```

```bash
NAME                                                                      READY  SEVERITY  REASON                       SINCE  MESSAGE
Cluster/private-external-8620                                             False  Warning   ScalingUp                    18m    Scaling up control plane to 3 replicas (actual 1)
├─ClusterInfrastructure - AzureCluster/private-external-8620              True                                          18m
├─ControlPlane - KubeadmControlPlane/private-external-8620-control-plane  False  Warning   ScalingUp                    18m    Scaling up control plane to 3 replicas (actual 1)
│ └─Machine/private-external-8620-control-plane-9rp2g                     True                                          15m
└─Workers
  └─MachineDeployment/private-external-8620-md-0                          False  Warning   WaitingForAvailableMachines  21m    Minimum availability requires 2 replicas, current 0 available
    └─2 Machines...                                                       True                                          13m    See private-external-8620-md-0-7cbcd647f-9vqs8, private-external-8620-md-0-7cbcd647f-hjg8n
```

> To fix the "False" READY status, [Azure cloud provider components](https://github.com/kubernetes-sigs/cloud-provider-azure/tree/master/helm/cloud-provider-azure) needs to be installed by helm.

Reviewer: ```suggestion
To fix the "False" READY status, [Azure cloud provider components](https://github.com/kubernetes-sigs/cloud-provider-azure/tree/master/helm/cloud-provider-azure) need to be installed by Helm.
```
